cluster-autoscaler/core/scaledown/planner/planner.go

```diff
@@ -285,18 +285,21 @@ func (p *Planner) categorizeNodes(podDestinations map[string]bool, scaleDownCand
 		p.unremovableNodes.Add(n)
 	}
 	p.nodeUtilizationMap = utilizationMap
+	timer := time.NewTimer(p.context.ScaleDownNodeCategorizationTimeout)
 	for _, node := range currentlyUnneededNodeNames {
-		// TODO(x13n): break on timeout. Figure out how to handle nodes
-		// identified as unneeded in previous iteration, but now
-		// skipped due to timeout.
-		removable, unremovable := p.rs.SimulateNodeRemoval(node, podDestinations, p.latestUpdate, pdbs)
-		if unremovable != nil {
-			unremovableCount += 1
-			p.unremovableNodes.AddTimeout(unremovable, unremovableTimeout)
-		}
-		if removable != nil {
-			delete(podDestinations, removable.Node.Name)
-			removableList = append(removableList, *removable)
+		select {
+		case <-timer.C:
```

Comment:
I think it would be useful to log a warning here: `%d out of %d nodes skipped in scale down simulation due to timeout`.

Reply:
added a warning